src/notifications/emails.py

```
@@ -0,0 +1,119 @@
+
+        try:
+            smtp = aiosmtplib.SMTP(hostname=self._hostname, port=self._port, start_tls=self._use_tls)
+            await smtp.connect()
+            if self._use_tls:
+                await smtp.starttls()
+            await smtp.login(self._email, self._password)
+            await smtp.sendmail(self._email, [recipient], message.as_string())
+            await smtp.quit()
+        except aiosmtplib.SMTPException as error:
+            logging.error(f"Failed to send email to {recipient}: {error}")
+            raise BaseEmailError(f"Failed to send email to {recipient}: {error}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)